src/Traits/Enqueue.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+namespace Backpack\Base\Traits;
+
+trait Enqueue
+{
+    public static $enqueuedScripts = [];
+    public static $enqueuedStyles = [];
+
+    public static function enqueueAsset($type, $id, $source, $dependencies = [])
+    {
+        switch ($type) {
+            case 'css':
+            case 'style':
+                return self::enqueueStyle($id, $source, $dependencies);
+                break;
+            case 'script':
+            case 'javascript':
+            case 'js':
+                return self::enqueueScript($id, $source, $dependencies);
+                break;
+        }
+    }
+
+    public static function enqueueScript($id, $source, $dependencies)
+    {
+        if (!str_contains($source, '//')) {
+            $path = '/'.ltrim($source, '/');
+            $v = filemtime(realpath(ltrim($path, '/')));
+            $sourceComputed = asset($path.'?v='.$v);
+        } else {
+            $sourceComputed = $source;
+        }
+
+        $sourceObj = (object) [
+            'id'     => $id,
+            'source' => $sourceComputed,
+            'deps'   => $dependencies,
+        ];
+
+        self::$enqueuedScripts[$id] = $sourceObj;
+
+        return $sourceObj;
+    }
+
+    public static function enqueueStyle($id, $source, $dependencies)
+    {
+        if (!str_contains($source, '//')) {
+            $path = '/'.ltrim($source, '/');
+            $v = filemtime(realpath(ltrim($path, '/')));
+            $sourceComputed = asset($path.'?v='.$v);
+        } else {
+            $sourceComputed = $source;
+        }
+
+        $sourceObj = (object) [
+            'id'     => $id,
+            'source' => $sourceComputed,
+            'deps'   => $dependencies,
+        ];
+
+        self::$enqueuedStyles[$id] = $sourceObj;
+
+        return $sourceObj;
+    }
+
+    private static function handleDependencies($items)
+    {
+        $items = array_unique($items, SORT_REGULAR); //Strip out duplicate requests
```

Comment:
What if it's NOT an array? i.e. should we validate that $items? Or at least give it a default value?